app/assets/javascripts/my_modules/results.js

```diff
@@ -89,117 +65,113 @@
-            parseInt($(table).css('max-height'), 10) - 30) {
-        $(table).find('.ht_master .wtHolder').css({ 'height': '100%',
-                                                     'width': '100%' });
-      }
+    // create custom ajax request in order to fix issuses with remote: true from
+    function handleResultFileSubmit(form, ev) {
+      ev.preventDefault();
+      ev.stopPropagation();
+
+      const url = form.find('#result_asset_attributes_file').data('directUploadUrl');
+      const file = form.find('#result_asset_attributes_file')[0].files[0];
+      const upload = new ActiveStorage.DirectUpload(file, url);
+
+      animateSpinner();
+
+      upload.create((error, blob) => {
+        if (error) {
+          // Handle the error
+        } else {
+          let formData = new FormData();
+          formData.append('result[name]', form.find('#result_name').val());
+          formData.append('result[asset_attributes][id]', form.find('#result_asset_attributes_id').val());
+          formData.append('result[asset_attributes][signed_blob_id]', blob.signed_id);
+
+          $.ajax({
+            type: 'PUT',
+            url: form.attr('action'),
+            data: formData,
+            success: function(data) {
+              animateSpinner(null, false);
+              $('.edit_result').parent().remove();
+              $(data.html).prependTo('#results').promise().done(() => {
+                $.each($('#results').find('.result'), function() {
+                  initFormSubmitLinks($(this));
+                  ResultAssets.applyEditResultAssetCallback();
+                  applyCollapseLinkCallBack();
+                  applyCreateWopiFileCallback();
+                  toggleResultEditButtons(true);
+                  FilePreviewModal.init();
+                  Comments.init();
+                  ResultAssets.initNewResultAsset();
+                  expandResult($(this));
+                });
+              });
+
+              $('#results-toolbar').show();
+            },
+            error: function(XHR) {
+              animateSpinner(null, false);
+              $('.edit_result').renderFormErrors('result', XHR.responseJSON.errors);
+            },
+            processData: false,
+            contentType: false
+          });
+        }
+      });
     }
 
-    function processResult(ev, resultTypeEnum, editMode) {
+    function processResult(ev, resultTypeEnum) {
       var $form = $(ev.target.form);
       $form.clearFormErrors();
 
-      switch(resultTypeEnum) {
+      switch (resultTypeEnum) {
         case ResultTypeEnum.FILE:
-          _handleResultFileSubmit($form, ev);
+          handleResultFileSubmit($form, ev);
           break;
         case ResultTypeEnum.TABLE:
-          var $nameInput = $form.find('#result_name');
-          var nameValid = textValidator(ev, $nameInput, 0,
-            <%= Constants::NAME_MAX_LENGTH %>);
+          textValidator(ev, $form.find('#result_name'), 0, $form.data('name-max-length'));
```

Comment:
Maybe we can use this constant file - https://github.com/biosistemika/scinote-web/blob/activestorage_migration/app/assets/javascripts/sitewide/constants.js.erb